index.js

```diff
@@ -51,44 +51,104 @@ var simplify = require('simplify-js');
+    feature.geometries = feature.geometries.map(function (g) {
+      if (supportedTypes.indexOf(g.type) > -1) {
+        simplified = simplifyHelper({
+          type: 'Feature',
+          geometry: g
+        }, tolerance, highQuality);
+
+        return simplified; // GeometryCollection shouldn't have properties
+      }
+      return g;
+    });
+
+    return feature;
+  } else {
+    return feature;
+  }
+};
+
+// supported GeoJSON geometries, used to check whether to wrap in simpleFeature()
+var supportedTypes = ['LineString', 'MultiLineString', 'Polygon', 'MultiPolygon'];
+
+function simplifyHelper (feature, tolerance, highQuality) {
   if(feature.geometry.type === 'LineString') {
     var line = {
       type: 'LineString',
+      coordinates: simplifyLine(feature.geometry.coordinates, tolerance, highQuality)
+    };
+
+    return line;
+  } else if(feature.geometry.type === 'MultiLineString') {
+    var multiline = {
+      type: 'MultiLineString',
       coordinates: []
     };
-    var pts = feature.geometry.coordinates.map(function(coord) {
-      return {x: coord[0], y: coord[1]};
-    });
-    line.coordinates = simplify(pts, tolerance, highQuality).map(function(coords) {
-      return [coords.x, coords.y];
+    // simplify each of the lines in the MultiLineString
+    feature.geometry.coordinates.forEach(function(lines) {
+      multiline.coordinates.push(simplifyLine(lines, tolerance, highQuality));
     });
 
-    return simpleFeature(line, feature.properties);
+    return multiline;
   } else if(feature.geometry.type === 'Polygon') {
     var poly = {
       type: 'Polygon',
+      coordinates: simplifyPolygon(feature.geometry.coordinates, tolerance, highQuality)
+    };
+
+    return poly;
+  } else if(feature.geometry.type === 'MultiPolygon') {
+    var multipoly = {
+      type: 'MultiPolygon',
       coordinates: []
     };
-    feature.geometry.coordinates.forEach(function(ring) {
-      var pts = ring.map(function(coord) {
-        return {x: coord[0], y: coord[1]};
-      });
-      var simpleRing = simplify(pts, tolerance, highQuality).map(function(coords) {
-        return [coords.x, coords.y];
-      });
-      for (var i = 0; i < 4; i++) {
-        if (!simpleRing[i]) simpleRing.push(simpleRing[0])
-      }
-      if (
-        (simpleRing[simpleRing.length-1][0] !== simpleRing[0][0]) ||
-        (simpleRing[simpleRing.length-1][1] !== simpleRing[0][1])) {
-        simpleRing.push(simpleRing[0])
-      }
-      poly.coordinates.push(simpleRing);
+    // simplify each set of rings in the MultiPolygon
+    feature.geometry.coordinates.forEach(function(rings) {
+      multipoly.coordinates.push(simplifyPolygon(rings, tolerance, highQuality));
     });
-    return simpleFeature(poly, feature.properties);
+
+    return multipoly;
+  } else {
+    // unsupported geometry type supplied
+    return feature;
   }
-};
+}
+
+/*
+* returns true if ring is a triangle
+*/
+function checkTriangle(ring) {
```

Comment:
I'd vote for renaming to `checkValidity` or something. Reading `checkTriangle` down below was misleading; I spent a few minutes trying to figure out why you would want to force a simplified polygon to be a triangle